src/dwd_dynam.jl

```diff
@@ -144,7 +176,7 @@ The readout function `r` may depend on the state, parameters and time, so it mus
 If it is left out, then ``r=u``.
 """
 const ContinuousMachine{T,I} = Machine{T, I, ContinuousDirectedSystem{T}}
-const InstantaneousContinuousMachine{T} = Machine{T, InstantaneousDirectedInterface{T}, ContinuousDirectedSystem{T}}
+const InstantaneousContinuousMachine{T,I} = Machine{T, I, ContinuousDirectedSystem{T}}
```

Comment:
Check here that the given interface `I` is a subtype of `AbstractInstantaneousDirectedInterface{T}`

Reply:
Thanks @slibkind! Could you explain why Julia doesn't like this: `const InstantaneousContinuousMachine{T,I} where {T,I<:AbstractInstantaneousDirectedInterface{T}} = Machine{T, I, ContinuousDirectedSystem{T}}`?

Reply:
@slibkind I tried to implement something like `const InstantaneousContinuousMachine{T,I<: AbstractInstantaneousDirectedInterface{T}} = Machine{T, I, ContinuousDirectedSystem{T}}` but it ran into problems with the constructors as implemented. That is, the constructor typed as `{T,N}` complained when `N` was given as an `Int64` for the dimensions of the vectors on the ports, because it expected an interface. So I did the checking in the constructors.

I think this is a general problem we have to live with because when I tried to do some type checking for regular continuous machines as `const ContinuousMachine{T,I <: AbstractDirectedInterface{T}} = Machine{T, I, ContinuousDirectedSystem{T}}` the constructor typed as `{T,N}` also failed with the type error `ERROR: TypeError: in Type, in I, expected I<:AlgebraicDynamics.DWDDynam.AbstractDirectedInterface{Float64}, got a value of type Int64`.

Reply:
Yes, I agree that having the vector length as part of the type parameter has led to some clunky issues. I made note of this in Issue #113.